boards/xtensa/esp32/common/src/esp32_ssd1680.c

```diff
@@ -0,0 +1,189 @@
+  return true;
+}
+#endif
+
+#if defined(CONFIG_SSD1680_GPIO_PIN_RST) && (CONFIG_SSD1680_GPIO_PIN_RST>=0)
+static bool ssd1680_set_rst(bool state)
+{
+  esp32_gpiowrite(CONFIG_SSD1680_GPIO_PIN_RST, state);
+  return true;
+}
+#endif
+
+#if defined(CONFIG_SSD1680_GPIO_PIN_BUSY) && (CONFIG_SSD1680_GPIO_PIN_BUSY>=0)
+static bool ssd1680_check_busy(void)
+{
+  return esp32_gpioread(CONFIG_SSD1680_GPIO_PIN_BUSY);
+}
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct lcd_dev_s    *g_lcddev;
+struct ssd1680_priv_s g_ssd1680_priv =
+{
+#if defined(CONFIG_SSD1680_GPIO_PIN_PWR) && (CONFIG_SSD1680_GPIO_PIN_PWR >= 0)
+  .set_vcc = ssd1680_set_vcc,
+#else
+  .set_vcc = NULL,
+#endif
+#if defined(CONFIG_SSD1680_GPIO_PIN_RST)  && (CONFIG_SSD1680_GPIO_PIN_RST >= 0)
+  .set_rst = ssd1680_set_rst,
+#else
+  .set_rst = NULL,
+#endif
+#if defined(CONFIG_SSD1680_GPIO_PIN_BUSY) && (CONFIG_SSD1680_GPIO_PIN_BUSY >= 0)
+  .check_busy = ssd1680_check_busy,
+#else
+  .check_busy = NULL,
+#endif
+};
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: board_lcd_initialize
+ ****************************************************************************/
+
+int board_lcd_initialize(void)
+{
+  struct spi_dev_s *spi;
+
+  /* Initialize additional I/O for e-ink display */
+
+#if defined(CONFIG_SSD1680_GPIO_PIN_DTA_CMD) && \
+  (CONFIG_SSD1680_GPIO_PIN_DTA_CMD >= 0)
+  esp32_configgpio(CONFIG_SSD1680_GPIO_PIN_DTA_CMD, OUTPUT);
+#endif
+
+#if defined(CONFIG_SSD1680_GPIO_PIN_PWR) && (CONFIG_SSD1680_GPIO_PIN_PWR >= 0)
+  esp32_configgpio(CONFIG_SSD1680_GPIO_PIN_PWR, OUTPUT);
+  lcdinfo("Using pin %d as PWR control\n", CONFIG_SSD1680_GPIO_PIN_PWR);
+#else
+  lcdinfo("PWR control line is disabled\n");
+#endif
+#if defined(CONFIG_SSD1680_GPIO_PIN_RST) && (CONFIG_SSD1680_GPIO_PIN_RST >= 0)
+  esp32_configgpio(CONFIG_SSD1680_GPIO_PIN_RST, OUTPUT);
+  lcdinfo("Using pin %d as RESET\n", CONFIG_SSD1680_GPIO_PIN_RST);
+#elif
+  lcdinfo("RESET line is disabled\n");
+#endif
+
+#if defined(CONFIG_SSD1680_GPIO_PIN_BUSY) && \
+  (CONFIG_SSD1680_GPIO_PIN_BUSY >= 0)
+  esp32_configgpio(CONFIG_SSD1680_GPIO_PIN_BUSY, INPUT | PULLUP);
+  lcdinfo("Using pin %d for reading busy state\n",
+    CONFIG_SSD1680_GPIO_PIN_BUSY);
+#elif
+  lcdinfo("Read busy line is disabled\n");
+#endif
+
+  /* Initialize SPI */
+
+  spi = esp32_spibus_initialize(CONFIG_SSD1680_SPI_BUS);
+  if (!spi)
+    {
+      lcderr("ERROR: Failed to initialize SPI port %d\n",
+        CONFIG_SSD1680_SPI_BUS);
+      return -ENODEV;
+    }
+  else
+    {
+      lcdinfo("Using SPI bus %d. SPI is initialized\n",
+        CONFIG_SSD1680_SPI_BUS);
```

Comment:
```suggestion
      lcdinfo("Using SPI bus %d. SPI is initialized\n",
              CONFIG_SSD1680_SPI_BUS);
```

Reply:
OK. I have applied suggestions. Thx. for Your remark.